src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java

```diff
@@ -1,83 +1,66 @@
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-import org.photonvision.EstimatedRobotPose;
-import org.photonvision.PhotonCamera;
-import org.photonvision.PhotonPoseEstimator;
 
 import java.util.Optional;
 
 
 public class Limelight extends GBSubsystem {
-	private static Limelight instance;
-	private PhotonCamera camera;
-	private PhotonPoseEstimator poseEstimator;
-	
-	private Limelight() {
-		camera = new PhotonCamera("photonvision");
-		poseEstimator = new PhotonPoseEstimator(
-				Field.Apriltags.aprilTagFieldLayout,
-				PhotonPoseEstimator.PoseStrategy.AVERAGE_BEST_TARGETS,
-				camera,
-				RobotMap.Vision.ROBOT_TO_CAMERA
-		);
-	}
-	
-	public static Limelight getInstance() {
-		if (instance == null) {
-			init();
-			SmartDashboard.putBoolean("limelight initialized via getinstance", true);
-		}
-		return instance;
-	}
-
-	public static void init(){
-		instance = new Limelight();
-	}
-	
-	
-	
-	/**
-	 * @return an array of three dimensions [x - forward , y - left, z - up]
-	 */
-	public Transform3d targetPos() {
-		var result = camera.getLatestResult();
-		if (!result.hasTargets()) {
-			return new Transform3d();
-		}
-		Transform3d target = result.getBestTarget().getBestCameraToTarget();
-		return target;
-	}
-	
-	/**
-	 * @return the estimated time the frame was taken
-	 */
-	public double getTimeStamp() {
-		var result = camera.getLatestResult();
-		return result.getTimestampSeconds();
-	}
-	public Optional<EstimatedRobotPose> getUpdatedPoseEstimator() {
-		return poseEstimator.update();
-	}
-	
-	public boolean hasTarget() {
-		return getUpdatedPoseEstimator().isPresent();
-	}
-	
-	/**
-	 * @return the apriltag id
-	 */
-	public int findTagId() {
-		if (hasTarget()) {
-			return camera.getLatestResult().getBestTarget().getFiducialId();
-		} else {
-			return 0;
-		}
-	}
-	
+    private static Limelight instance;
+    private NetworkTableEntry robotPoseEntry, jsonEntry, idEntry;
+    public final static double MIN_DISTANCE_TO_FILTER_OUT = 1;
+
+    private Limelight() {
+        String robotPoseQuery = DriverStation.getAlliance() == DriverStation.Alliance.Blue ? "botpose_wpiblue" : "botpose_wpired";
+        robotPoseEntry = NetworkTableInstance.getDefault().getTable(RobotMap.Vision.LIMELIGHT_NAME).getEntry(robotPoseQuery);
+        jsonEntry = NetworkTableInstance.getDefault().getTable(RobotMap.Vision.LIMELIGHT_NAME).getEntry("json");
+        idEntry = NetworkTableInstance.getDefault().getTable(RobotMap.Vision.LIMELIGHT_NAME).getEntry("tid");
+    }
+
+    public static Limelight getInstance() {
+        if (instance == null) {
+            init();
+            SmartDashboard.putBoolean("limelight initialized via getinstance", true);
+        }
+        return instance;
+    }
+
+    public static void init(){
+        instance = new Limelight();
+    }
+
+
+    public Optional<Pair<Pose2d, Double>> getUpdatedPoseEstimator() {
+        double[] poseArray = robotPoseEntry.getDoubleArray(new double[7]);
```

Comment:
magic nums, magic nums everywhere

Reply:
added doc do not think magic numbers are avoidable

Reply:
at least not the indices

Reply:
doc is good enough